### DISCUSSION TITLE
A plea for help

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dotnet/maui#14680] [General]
### Description

Hi. This is a plea for help...

We have been battling to get a Maui app live for some time. It is impossible due to some bugs we are battling with, so I am pleading/begging with you to see if there is any way someone can look at addressing our list of bugs so that we can progress... 

I know this is probably an unusual request but we have invested a lot of time in working through these issues and raising them with you (it consumes a lot of time to put working examples together - and this comes after losing weeks trying to work out if it is our lack of understanding or a Maui bug). To date only 1 of the issues we raised has been looked at (although no sign of it being released yet).

We have put everything into Maui, but so far we are running at least 6 months behind due to all the issues, and at the current rate of problems being addresses I can see the customer cancelling the project before we get to having a release.

So I am asking/begging - can someone look at these bugs (in priority order)?

The first 4 look like a similar root cause:
https://github.com/dotnet/maui/issues/14606 
https://github.com/dotnet/maui/issues/14118
https://github.com/dotnet/maui/issues/13893
https://github.com/dotnet/maui/issues/14526

These are also high priority:
https://github.com/dotnet/maui/issues/14284
https://github.com/dotnet/maui/issues/13695

Lower priority:
https://github.com/dotnet/maui/issues/14649
https://github.com/dotnet/maui/issues/14120
https://github.com/dotnet/maui/issues/14651
https://github.com/dotnet/maui/issues/13981
https://github.com/dotnet/maui/issues/14173

This one has also been raised with the VS team, but no progress on either request. It is a blocker for iOS development because it makes VS for Mac effectively unusable:
https://github.com/dotnet/maui/issues/14175

This one has been marked as done but is still open:
https://github.com/dotnet/maui/issues/14052

I am happy to do whatever is needed to get these looked at. You can have access to our repository so you can see the problems first hand which would probably give you some insight to real world scenarios...

Thanks

### Steps to Reproduce

N/A

### Link to public reproduction project repository

https://github.com/williambuchanan2/MauiNavigation

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android

### Affected platform versions

Android 11 - 13, iOS 16

### Did you find any workaround?

No

**Comment by HuaFangYun** [comment]
Maui needs to speed up updates and bug fixes, which is a real problem for now, I think.

**Reply by janseris** [reply]
It's been super slow since its launch (May/June 2022), I've written about it to their project manager, warned in live official MAUI (dotnet channel) YouTube videos, on a lot of GitHub issues and nothing has changed (yea, Microsoft-employed testers disappeared from GitHub issues :) )
An intriguing thing is that new basic bugs are introduced in new service releases.
It was funny to watch that in one service release in .NET 6, even the basic MAUI project template was not working correctly (it contains just a simple image and a button) and that service release was still released into production.
This gives up a very nice insight about the overall project quality.

**Reply by Xyncgas** [reply]
What's your youtube

**Reply by KarthikHebbarMN** [reply]
> It's been super slow since its launch (May/June 2022), I've written about it to their project manager, warned in live official MAUI (dotnet channel) YouTube videos, on a lot of GitHub issues and nothing has changed (yea, Microsoft-employed testers disappeared from GitHub issues :) ) An intriguing thing is that new basic bugs are introduced in new service releases. It was funny to watch that in one service release in .NET 6, even the basic MAUI project template was not working correctly (it contains just a simple image and a button) and that service release was still released into production. This gives up a very nice insight about the overall project quality.

Exactly, they prematurely depricated Xamrin which was active for 9 years. 

It has frustrated me so much that, I feel like dropping all dot net projects and switch to Flutter and Python/Go

**Reply by Xyncgas** [reply]
Use F#, it's more portable than C#, if you are moving away from .NET you can use python javascript packages instead of nuget

**Comment by ScarletKuro** [comment]
I don't want to offend anyone, but it seems that your risk assessment may be inaccurate. It's important to carefully assess the risks of adopting new technologies, especially when it comes to critical projects. MAUI was only released less than a year ago, and it will take at least 2 more years until it is mature enough and ready for production. This was the same situation with Blazor when it first appeared in .NET Core 3.0, and it had too many problems to be used in anything other than pet projects. While it may be tempting for .NET developers to use MAUI, there are currently better options for mobile development, such as Flutter. Even Xamarin would be a better choice right now, despite being replaced by MAUI. I have experience using Xamarin myself, and while it does have its issues, I still consider it to be more stable than MAUI at the moment. Additionally, there are many well-established community components that are stable and reliable, and workarounds can often be found for common issues. While MAUI certainly has potential, it's still a young technology and it may take some time before it's truly ready for production use. I hope that this comment will serve as a helpful guide for others in the future to avoid similar situations.

**Reply by janseris** [reply]
> I don't want to offend anyone, but it seems that your risk assessment may be inaccurate. It's important to carefully assess the risks of adopting new technologies, especially when it comes to critical projects. MAUI was only released less than a year ago, and it will take at least 2 more years until it is mature enough and ready for production. This was the same situation with Blazor when it first appeared in .NET Core 3.0, and it had too many problems to be used in anything other than pet projects. While it may be tempting for .NET developers to use MAUI, there are currently better options for mobile development, such as Flutter. Even Xamarin would be a better choice right now, despite being replaced by MAUI. I have experience using Xamarin myself, and while it does have its issues, I still consider it to be more stable than MAUI at the moment. Additionally, there are many well-established community components that are stable and reliable, and workarounds can often be found for common issues. While MAUI certainly has potential, it's still a young technology and it may take some time before it's truly ready for production use. I hope that this comment will serve as a helpful guide for others in the future to avoid similar situations.

You are right and I will only leave 1 thing here - MAUI team wrote on .NET Blog that the last Release Candidate in April/May 2022 was _production ready_ while MAUI isn't production ready even now.
One might get very easily confused by the official statements and official propagation of the MAUI technology by Microsoft and MAUI creators/PM/devs on YouTube and Microsoft websites.

**Reply by williambuchanan2** [reply]
> I don't want to offend anyone, but it seems that your risk assessment may be inaccurate. It's important to carefully assess the risks of adopting new technologies, especially when it comes to critical projects. MAUI was only released less than a year ago, and it will take at least 2 more years until it is mature enough and ready for production. This was the same situation with Blazor when it first appeared in .NET Core 3.0, and it had too many problems to be used in anything other than pet projects. While it may be tempting for .NET developers to use MAUI, there are currently better options for mobile development, such as Flutter. Even Xamarin would be a better choice right now, despite being replaced by MAUI. I have experience using Xamarin myself, and while it does have its issues, I still consider it to be more stable than MAUI at the moment. Additionally, there are many well-established community components that are stable and reliable, and workarounds can often be found for common issues. While MAUI certainly has potential, it's still a young technology and it may take some time before it's truly ready for production use. I hope that this comment will serve as a helpful guide for others in the future to avoid similar situations.

No offence taken, and you are absolutely right. I have to say it has been a lesson learned. After this, my confidence in Microsoft's competence is close to zero.

Choosing Maui over Flutter seems a mistake now. In my defence, we have to make these calls every day when looking at future tech, and based on the marketing and announcements from the Maui team - claiming that it was "prod ready" a year ago - we decided to go with this. Who would have known then what an utter disaster it really was. It was inevitable that there would be some bugs and challenges, but here is what has caught me out:

a. The sheer number of bugs. The phrase "production ready" has never been so misused. Who could have predicted that?
b. The complete lack of ownership in dealing with those bugs. I'm seeing critical issues that haven't moved in 18 months, and everything else either closed off instantly "unable to reproduce", or thrown into the backlog never to see the light of day again. One of those critical bugs I noticed the other day was eventually fixed - it was a 1 line change, yet it took 18 months.
c. Surprise that with an unusable Maui version in .NET 6, and now and unusable version in .NET 7, they have already abandoned us and moved their focus to .NET 8. Who in their right mind would reallocate people to the next version when they haven't even finished the 2 prior versions?
d. We have become the defacto testers, and first line support for Maui. We are finding all the bugs, and spending all our valuable time putting together working examples and sending it through to the second line support. I wouldn't have an issue with this if I got something in return.

IMHO this just looks like a complete failure in management. From what I can see of Maui, it is a work of genius, and there are clearly some very talented people involved in the development of it, but I would guess the handful of devs and testers on the project are stressed to the max trying to keep the plates spinning. You have to ask why management have let it fail - why haven't they loaded the team up with devs and testers, to get at least the most basic level of quality in place before moving the team on to the next big thing.

I'm not trying to offend or being abusive - i'm just saying what I see, and I am sure everyone else who has decided to stake their reputations in Maui is also thinking. I may be completely wrong and maybe everything is completely under control, but I think the example given about the click counter demo app being shipped with an early version not working kind of sums the whole thing up (in fact I spent some time trying to get that thing working, thinking the whole time I was doing something wrong).

But again - my only reason for raising this was to plead for someone to do something...

**Reply by Xyncgas** [reply]
The benefit of microsoft doing this is they don't have to deal with your customer, and they don't have to deal with you, although if you are successful they get a credit for it and they benefit further by pulling you in their eco system with nuget and azure and what not so be careful

**Reply by prolixmusic** [reply]
Yeah, I made the same mistake... I can't tell you how many hours I have wasted trying to find workarounds for things that don't work the way they are supposed to...  VisualStates dont work properly unless you include the <VisualState x:Name="PointerOver"/> which incidentally isn't in any of the precooked styles. Also, IsEnabled doesn't propagate.. It has no effect in TabbedPage, ContentPage, VerticalStackLayout, Grid, etc... So annoying!

**Reply by SJaschinski** [reply]
Cannot agree with you @ScarletKuro as Microsoft forces all Xamarin projects to move to MAUI asap. In general I would totally agree and wait for one or even tow more years before I migrate my project from Xamarin to MAUI. But please do not forget the official Microsoft roadmap regarding the end of support for Xamarin https://dotnet.microsoft.com/en-us/platform/support/policy/xamarin. This currently says that the support for Xamarin will end in May 2024 and even worse the Xamarin will not support any iOS version after 16 which de facto means that you won't be able to produce a Xamarin app for the latest iOS version from fall 2023 on. By the way this has just been confirmed again by the MAUI team at Build last week. This is a very limited timeframe given my experiences I had so far trying to migrate Xamarin to MAUI. But the risk of staying with Xamarin for now and have to deal with unsupported versions of iOS, XCode etc. in the near future is worse than at least try to make the move to MAUI if you already have a major investment as a company into a Xamarin app. Don't you think so too?

**Comment by manishbisen** [comment]
@davidortinau may help here.

**Comment by janseris** [comment]
Also a quick note about estimations on MAUI. 
How fast do you expect 2200 issues being fixed by ~10 MAUI devs if 1 year ago they had also 10 devs but 600 issues? The future and present looks very dark.

**Reply by williambuchanan2** [reply]
> Also a quick note about estimations on MAUI. How fast do you expect 2200 issues being fixed by ~10 MAUI devs if 1 year ago they had also 10 devs but 600 issues? The future and present looks very dark.

Are there 10 working on this? Looks to me like only 2 or 3, and a few others who just look at the bug reports and throw them in the backlog.

**Reply by janseris** [reply]
> > Also a quick note about estimations on MAUI. How fast do you expect 2200 issues being fixed by ~10 MAUI devs if 1 year ago they had also 10 devs but 600 issues? The future and present looks very dark.
> 
> Are there 10 working on this? Looks to me like only 2 or 3, and a few others who just look at the bug reports and throw them in the backlog.

I just counted 6 from memory so that 10 is a rough estimate for what I forgot or have missed in the commits and issues.

**Reply by Kalyxt** [reply]
I scrapped over two weeks of work on MAUI project and started fresh project in Flatter with zero knowledge. (Its more enjoyable experience tho)  It just doesn't worth the struggle, workaround after workaround to workaround. Another path could be to stick with Xamarin, they might not cut it so early when MAUI is joke right now.

**Comment by Ghevi** [comment]
Isn't Maui Blazor a solution for you at the moment? Too much to rewrite?
I've being developing and publishing an app using Blazor components and few Xaml pages when needed, it seems to be good enough even tho there are some weird bugs sometimes and you have to make it responsive with bootstrap for different screen sizes but whatever.

**Reply by williambuchanan2** [reply]
> Isn't Maui Blazor a solution for you at the moment? 

Too much rewrite I think... We have lost so much time on the project, and stuck with it because we thought the problems would be getting fixed, and now we are past the point of no return. I think to get the team trained up on Blazor and then work through all the problems it has wouldn't be feasible in the time we have left...

**Reply by ScarletKuro** [reply]
I wouldn't necessarily consider MAUI Blazor as the solution either. There have been many issues related to published applications(you can find them in issues). For example this is one of them that MudBlazor customers faced (https://github.com/dotnet/maui/issues/14271), and the solution is not always obvious or it's difficult to explore. Other problems, such as as this (https://github.com/dotnet/maui/issues/13529). Even adding something as basic as deep-linking could be a struggle (https://github.com/dotnet/maui/issues/3788).

**Reply by janseris** [reply]
> I wouldn't necessarily consider MAUI Blazor as the solution either. There have been many issues related to published applications(you can find them in issues). For example this is one of them that MudBlazor customers faced (#14271), and the solution is not always obvious or it's difficult to explore. Other problems, such as as this (#13529). Even adding something as basic as deep-linking could be a struggle (#3788).

And this:
https://github.com/dotnet/maui/issues/7277

**Reply by ScarletKuro** [reply]
> And this: #7277

Yes, this ones are painful.

**Reply by Xyncgas** [reply]
https://github.com/microsoft/WindowsAppSDK/issues/3266
You can't use mouse on .NET 7
Webview is bugged

**Reply by Xyncgas** [reply]
F# / C# is doing ok just don't trust microsoft's software solution, write your own, instead of bearing the technological debt a company should develop their own framework, nowadays relying on anyone gives them the power to take the thing you rely on away from you and ruin your company, be independent the effort you put into your own framework is going to be your asset

**Reply by Ghevi** [reply]
Ah damn, I don't use Maui Blazor for Windows or MudBlazor, nor JsInterops. For the DeepLink yeah it's a struggle but doable.
I saw him listing only iOS or Android. Regardless, what do you mean writing our own framework? Sorry im a junior dev.
You mean a UI framework from scratch??

**Comment by akhanalcs** [comment]
This project is starting to look like a joke.

**Reply by Xyncgas** [reply]
Still, I am kinda using it, so please don't kill it like windows phone

**Reply by williambuchanan2** [reply]
> Still, I am kinda using it, so please don't kill it like windows phone

I loved windows phone. I wish they had just stuck with it a bit longer. It was a much better OS than iOS or Android in my opinion!

**Reply by ScarletKuro** [reply]
> I loved windows phone. I wish they had just stuck with it a bit longer. It was a much better OS than iOS or Android in my opinion!

I completely agree with you. I also miss my Microsoft Lumia 950 and I think Windows Phone was a really interesting operating system that offered a unique and refreshing experience compared to Android and iOS. I particularly liked the Metro UI, and have even used it in some of my own projects. The "Continuum" feature was also really promising and I even invested in the Microsoft Display Dock, although unfortunately it's now somewhat useless. I had high hopes for Windows 10X, especially since they used the Lumia 950/XL as the SDK device, but unfortunately they announced that they were discontinuing the project.

**Reply by ScarletKuro** [reply]
> so please don't kill it

There have been rumors circulating that Microsoft Management is considering dropping the native aspect of MAUI and instead focusing solely on Blazor hybrid. This is in response to the massive amount of dissatisfaction from the community. While these are only rumors and the PM of MAUI has denied them, it is always possible that things could change. Ultimately, decisions made by Management carry more weight than those made by PMs.

**Reply by ziomek64** [reply]
> > so please don't kill it
> 
> There have been rumors circulating that Microsoft Management is considering dropping the native aspect of MAUI and instead focusing solely on Blazor hybrid. This is in response to the massive amount of dissatisfaction from the community. While these are only rumors and the PM of MAUI has denied them, it is always possible that things could change. Ultimately, decisions made by Management carry more weight than those made by PMs.
> 
> 

Source?

**Reply by ScarletKuro** [reply]
> Source?

Was on [Reddit](https://www.reddit.com/r/csharp/comments/122yelz/maui_going_to_all_in_on_blazor_hybrid/).  It's hard to call it a trustworthy source. Again, it's just rumors and speculations. PM response you can find in comments.

**Reply by janseris** [reply]
> > Source?
> 
> Was on [Reddit](https://www.reddit.com/r/csharp/comments/122yelz/maui_going_to_all_in_on_blazor_hybrid/). It's hard to call it a trustworthy source. Again, it's just rumors and speculations. PM response you can find in comments.

It's funny that the rumor says downsizing the team even more. Insane but at the same believable. 
How is it gonna end if the team is even smaller - the current devs don't do just the native UI part but the whole framework. MAUI Blazor will still have to use the "backend" stuff which is bugged as well (but does not usually break in newer releases).
For example icon and splashscreen was broken for more than 1 year in a row. No documented properties worked and no image other than using a PNG with the same file name as the hardcoded default.
It's not just the UI which is bad in MAUI.
Remember how MAUI could not read files, do Bluetooth or request permissions properly on Android 13 for multiple months.

**Reply by williambuchanan2** [reply]
> There have been rumors circulating that Microsoft Management is considering dropping the native aspect of MAUI and instead focusing solely on Blazor hybrid. 

That would be a really dumb move - and yet believable. We have started the process of standardising everything on .NET. I'm afraid if that rumour is true we will look at getting out of .NET completely. Just can't be doing with the instability and constant changing in direction. Blazor isn't really a solution. A hybrid app will only get you so far, so I think for us it would trigger a switch to Flutter, or just go back to native dev.

**Comment by ziomek64** [comment]
It looks like a joke a bit. I agree with lots of points made here. MAUI team is hugely understaffed and the project is going down. .NET 6 version is bugged, 7 is even more buggy and 8 is the buggiest of them all. Is it really necessary to move from .net version to .net version each time instead of patiently fixing one version for one more year? A lot of critical bugs are not fixed, but let's add new features in new .net version just for the sake of it. Same goes for toolkit, instead of going for feature parity with the old xamarin toolkit we get some new features. The most miserable example is the damn touch effect

I really like Xamarin and MAUI because it's easy to learn even if you are beginner developer it's much easier for me to understand xaml and c# instead of flutter and i'd like it to succeed

**Reply by williambuchanan2** [reply]
Well said...

**Reply by Xyncgas** [reply]
Look at it this way for now it's buggy, so if you can fix all your bugs that means you are more competitive on the market, if it takes 5 years for MAUI to be perfect in the future then you are 5 years ahead of other companies who's using MAUI now if you can figure out how to make your app working.

I just fixed all my bugs today and I think I am better than all of you

**Reply by Xyncgas** [reply]
But it's not easy to write MAUI.Blazor too, the advance scenario for building your own render tree and updating it was quite a read too

**Reply by janseris** [reply]
> I hope MAUI adopt Nuget package as Xamarin.Forms, as soon as possible!
> 
> MAUI is a complex architecture that work with Native platform features (layout for example), that have different implementations, but the same results.
> 
> So i expect issues, but with nuget it can be fixed, faster, without wait all visual studio build.
> 
> We can't wait months mostly, for hot issues, only because we must wait visual studio next release!

The fixes aren't slow because of Visual Studio releases.

**Reply by ziomek64** [reply]
> Look at it this way for now it's buggy, so if you can fix all your bugs that means you are more competitive on the market, if it takes 5 years for MAUI to be perfect in the future then you are 5 years ahead of other companies who's using MAUI now if you can figure out how to make your app working.
> 
> I just fixed all my bugs today and I think I am better than all of you

Our bugs? I don't intend to fix MAUI bugs, I just want relatively bugless environment.

**Reply by ziomek64** [reply]
> > I hope MAUI adopt Nuget package as Xamarin.Forms, as soon as possible!
> > MAUI is a complex architecture that work with Native platform features (layout for example), that have different implementations, but the same results.
> > So i expect issues, but with nuget it can be fixed, faster, without wait all visual studio build.
> > We can't wait months mostly, for hot issues, only because we must wait visual studio next release!
> 
> The fixes aren't slow because of Visual Studio releases.

They definitely aren't. As soon as new maui version drops, new visual studio drops in max week

**Comment by GlacierFox** [comment]
It's kinda annoying that your issues seem to have been closed for no apparent reason. That image size issue seems like a critical problem for anyone hoping to release on IOS.

**Reply by Xyncgas** [reply]
In my experience, developing for Blazor and hearing Blazor is available in MAUI, sticking to blazor has caused the least friction, I was developing framework for it, despite .NET 5 updated to .NET 6 to .NET 7, I can continue doing what I am doing

**Reply by williambuchanan2** [reply]
> That image size issue seems like a critical problem for anyone hoping to release on IOS.

You are absolutely right. It's hard to imagine an app without some kind of image in it, and yet you can't reliably have an image in iOS with Maui. So I would call it critical.

**Reply by ziomek64** [reply]
#14471 this one is even worse, I can't use .net 7 because of this lol. Such critical issue, 3 weeks untouched already

**Reply by williambuchanan2** [reply]
> #14471 this one is even worse, I can't use .net 7 because of this lol. Such critical issue, 3 weeks untouched already

Ah that could be similar to another problem I am seeing. When scrolling a page which has a lot of images some of them go blank just like that. But 3 weeks is nothing. I doubt it will be fixed for at least 18 months.

**Reply by ziomek64** [reply]
Well they should start prioritizing such stuff because it makes Maui COMPLETELY unusuable, it's not a small issue that you can workaround at all

**Reply by williambuchanan2** [reply]
You are right. At the moment it would be very hard to actually deliver anything but the most basic of app. I think most of the bugs can be worked around though so as long as they fix the major ones then it will be a very good platform to use - but there is no sign of that happening!

**Comment by HuaFangYun** [comment]
It seems that since last week, the update speed has become faster. I hope it is a good start.

**Comment by ziomek64** [comment]
> ### Description
> 
> Hi. This is a plea for help...
> 
> We have been battling to get a Maui app live for some time. It is impossible due to some bugs we are battling with, so I am pleading/begging with you to see if there is any way someone can look at addressing our list of bugs so that we can progress... 
> 
> I know this is probably an unusual request but we have invested a lot of time in working through these issues and raising them with you (it consumes a lot of time to put working examples together - and this comes after losing weeks trying to work out if it is our lack of understanding or a Maui bug). To date only 1 of the issues we raised has been looked at (although no sign of it being released yet).
> 
> We have put everything into Maui, but so far we are running at least 6 months behind due to all the issues, and at the current rate of problems being addresses I can see the customer cancelling the project before we get to having a release.
> 
> So I am asking/begging - can someone look at these bugs (in priority order)?
> 
> The first 4 look like a similar root cause:
> https://github.com/dotnet/maui/issues/14606 
> https://github.com/dotnet/maui/issues/14118
> https://github.com/dotnet/maui/issues/13893
> https://github.com/dotnet/maui/issues/14526
> 
> These are also high priority:
> https://github.com/dotnet/maui/issues/14284
> https://github.com/dotnet/maui/issues/13695
> 
> Lower priority:
> https://github.com/dotnet/maui/issues/14649
> https://github.com/dotnet/maui/issues/14120
> https://github.com/dotnet/maui/issues/14651
> https://github.com/dotnet/maui/issues/13981
> https://github.com/dotnet/maui/issues/14173
> 
> This one has also been raised with the VS team, but no progress on either request. It is a blocker for iOS development because it makes VS for Mac effectively unusable:
> https://github.com/dotnet/maui/issues/14175
> 
> This one has been marked as done but is still open:
> https://github.com/dotnet/maui/issues/14052
> 
> I am happy to do whatever is needed to get these looked at. You can have access to our repository so you can see the problems first hand which would probably give you some insight to real world scenarios...
> 
> Thanks
> 
> ### Steps to Reproduce
> 
> N/A
> 
> ### Link to public reproduction project repository
> 
> https://github.com/williambuchanan2/MauiNavigation
> 
> ### Version with bug
> 
> 7.0 (current)
> 
> ### Last version that worked well
> 
> Unknown/Other
> 
> ### Affected platforms
> 
> iOS, Android
> 
> ### Affected platform versions
> 
> Android 11 - 13, iOS 16
> 
> ### Did you find any workaround?
> 
> No
> 
> ### Relevant log output
> 
> _No response_

Would pretty cool to finally have MAUI as nugets. So we could compile it for ourselves or get those nighty/weekly builds to have the most bleeding edge experience, it would speed up things a lot instead of waiting. What do you guys think

**Reply by nll** [reply]
There are steps to build and use a particular version of MAUI locally (not sure if it still works): https://dev.to/davidortinau/using-the-latest-net-maui-2f88

From what I understand MAUI is (in part) moving away from workloads and more into nugets, and the way to use a temporary version using `MauiVersion` is a temporary solution, so the in the future will be easier.

 - https://github.com/dotnet/maui/pull/11206
 - https://github.com/dotnet/maui/pull/11946
 - https://github.com/dotnet/maui/issues/12998#issuecomment-1420484976
 - https://github.com/dotnet/docs-maui/issues/1286

**Comment by ziomek64** [comment]
After a month of waiting we got an update, 3 bug fixes lol. Are they preparing for .net 8 or what.

**Reply by williambuchanan2** [reply]
Did 3 bugs get fixed? I missed that. Unfortunately none of the ones I raised were looked at :(

**Reply by ziomek64** [reply]
> Did 3 bugs get fixed? I missed that. Unfortunately none of the ones I raised were looked at :(

Yeah, that's what it says. And i wonder why because last month there was a massive list of fixes on both .net 7 and 8

**Reply by HuaFangYun** [reply]
I've been keeping an eye on maui updates, but a lot of updates are optimized rather than dealing with bug. Maybe after optimizing to a certain program later, bug will be repaired in large quantities.

**Reply by ziomek64** [reply]
> I've been keeping an eye on maui updates, but a lot of updates are optimized rather than dealing with bug. Maybe after optimizing to a certain program later, bug will be repaired in large quantities.

I think they should fix bugs first then optimize the whole thing to refine it

**Comment by Xyncgas** [comment]
It's been a month, how's your tea

**Reply by ziomek64** [reply]
Cold

**Comment by janseris** [comment]
Nothing will change, guys. The support hasn't changed for 1 year, what do you expect?

Because noone from MAUI team reacted to this, I will try instead:

**We would like to take this opportunity to update you on our progress with .NET MAUI. Our team is fully committed to enhancing the quality and performance of the framework while continuously delivering new functionality. We understand the importance of these aspects and are actively working on them.**

**We value your feedback and want to assure you that your reported issues are receiving our utmost attention. While we are diligently addressing the reported issues, we also acknowledge that there may be instances where the resolution takes longer than expected. Please know that we are actively working on all reported issues, and your patience is greatly appreciated.**

**We remain dedicated to providing you with a superior experience and appreciate your ongoing support.**

**Sincerely,
MAUI Team Leader's Name**

Enjoy

**Reply by williambuchanan2** [reply]
Ha that is hillarious. I think if they did post a response like that we would be justified to call them dillusional!

**Comment by christosk92** [comment]
Hindsight is 20/20 but maybe you should've considered Avalonia.

**Reply by williambuchanan2** [reply]
Yeah I think with 20/20 vision i'd have basically chosen anything except Maui...

**Reply by Xyncgas** [reply]
> Hindsight is 20/20 but maybe you should've considered Avalonia.

Are you a fan of Avalonia, how are you having it

**Reply by christosk92** [reply]
> > Hindsight is 20/20 but maybe you should've considered Avalonia.
> 
> Are you a fan of Avalonia, how are you having it

Really stable, very performant, consistent releases/quick release schedule.
Only have good words really.

**Reply by robloo** [reply]
Avalonia is a no-brainer for ANY cross-platform desktop applications right now (macOS, Windows, Linux). It is already used in production for some very complex apps. Their website gives good examples: https://avaloniaui.net/Showcase.

Within a year I expect Avalonia will also be a no-brainer for mobile apps on iOS and Android. It's performance there is expected to be on par with Flutter and far exceeding Maui.

A complete comparison made for business decisions was released publicly here: https://github.com/robloo/PublicDocs/blob/master/XAMLFrameworkComparison.md. That set the strategy on my end and Avalonia was the clear choice to move to following the UWP debacle (and WinUI 3 will fail similarly: https://github.com/microsoft/microsoft-ui-xaml/issues/3639). If your app can launch with desktop versions first I would recommend starting a re-write now. Going from Maui to Avalonia is going to be day vs. night difference and you will save a lot of time/money in the long run.

Maui was also destined to fail from the beginning. @davidortinau failed to understand the discussion here: https://github.com/dotnet/maui/issues/43. I even called out explicitly the bug situation. They never learn.

Microsoft has lost all vision in the UI framework space.

**Reply by timahrentlov** [reply]
> Avalonia is a no-brainer for ANY cross-platform desktop applications right now (macOS, Windows, Linux). 

I agree. Looking forward to the v11 docs getting released though.
